storage/badger/cleaner.go

```diff
@@ -10,78 +10,95 @@ import (
 	"github.com/rs/zerolog"
 
 	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
 )
 
+// Cleaner uses component.ComponentManager to implement module.Startable and module.ReadyDoneAware
+// to run an internal goroutine which run badger value log garbage collection on timely basis.
```

Comment:
```suggestion
// to run an internal goroutine which run badger value log garbage collection at a semi-regular interval.
// The Cleaner exists for 2 reasons:
//   - Run GC frequently enough that each GC is relatively inexpensive
//   - Avoid GC being synchronized across all nodes. Since in the happy path, all nodes have very similar 
//     database load patterns, without intervention they are likely to schedule GC at the same time, which 
//     can cause temporary consensus halts.
```